natlib/svcd.c

```diff
@@ -183,5 +190,38 @@ static int svcd_init( lua_State *L )
         lua_call(L, 3, 0);
     }
 
+    
+    return 0;
+}
+
+// Lua: storm.n.ndispatch ( pay, srcip, srcport )
+// Dispatches an incoming notification
+
+static int svcd_ndispatch( lua_State *L )
+{
+    // Checks to make sure it is passed 3 parameters
+    if (lua_gettop(L) != 3) return luaL_error(L, "Expected (pay, srcip, srcport)"); 
+
+    size_t parlen;
+    const char* pay = lua_tolstring(L, -1, &parlen); 
+    uint16_t ivkid = lua_tonumber(L, -1); // changes pay parameter to number and returns
+    lua_getglobal(L, "SVCD"); // gets table
+    lua_pushstring(L, "oursubs"); // use oursubs as the key
+    lua_gettable(L, 5);
```

Comment:
lua_gettable is expects the index of the table to be provided as an argument, and pops the key from the top of the stack. Unless I'm mistaken the table is at index 4 and the key is at index 5. So to get SVCD.oursubs, you'd do lua_gettable(L, 4), which would pop "oursubs" from the stack and then push SVCD.oursubs (which you want) to the top.